File: app/src/main/java/com/absinthe/kage/device/cmd/SetPlayStatusCommand.java
package com.absinthe.kage.device.cmd;

import com.absinthe.kage.connect.protocol.IpMessageConst;
import com.absinthe.kage.device.Command;
import com.absinthe.kage.device.CommandBuilder;
import com.absinthe.kage.device.client.Client;

public class SetPlayStatusCommand extends Command {

    public int statusCode;

    public SetPlayStatusCommand() {
        cmd = IpMessageConst.MEDIA_SET_PLAYER_STATUS;
    }

    @Override
    public String pack() {
        return new CommandBuilder()
                .with(this)
                .append(String.valueOf(statusCode))
                .build();
    }

    @Override
    public void doWork(Client client, String received) {

    }

    @Override
    public boolean parseReceived(String received) {
        return false;
    }
}

File: app/src/main/java/com/absinthe/kage/device/cmd/SetPlayStatusCommand.kt
package com.absinthe.kage.device.cmd

import com.absinthe.kage.connect.protocol.IpMessageConst
import com.absinthe.kage.device.Command
import com.absinthe.kage.device.CommandBuilder
import com.absinthe.kage.device.client.Client

class SetPlayStatusCommand : Command() {
    @JvmField
    var statusCode = 0

    override fun pack(): String {
        return CommandBuilder()
                .with(this)
                .append(statusCode.toString())
                .build()
    }

    override fun doWork(client: Client, received: String) {}

    override fun parseReceived(received: String): Boolean {
        return false
    }

    init {
        cmd = IpMessageConst.MEDIA_SET_PLAYER_STATUS
    }
}